spec/presenters/concerns/candidate_api_data_spec.rb

```diff
@@ -96,6 +96,17 @@ def initialize(application_choice)
           expect(presenter.candidate[:uk_residency_status]).to eq('Candidate needs to apply for permission to work and study in the UK')
         end
       end
+
+      context 'when the right to work or study details go over the character limit' do
+        details = Faker::Lorem.characters(number: 257)
+        truncated_details = details.truncate(256, omission: '... (this field was truncated as it went over the length limit)')
+
+        let(:application_form) { create(:application_form, :minimum_info, first_nationality: 'Canadian', right_to_work_or_study: 'yes', right_to_work_or_study_details: details) }
+
+        it 'returns a truncated message' do
+          expect(presenter.candidate[:uk_residency_status]).to eq(truncated_details)
```

Comment:
`truncated_details` are set in a constant which we can access and compare through `.to end_with(described_class::OMISSION_TEXT)` instead of defining again. Also can you add a test that checks the total value of chars is 256, as well as that we raise a Sentry error?

Reply:
Done!